ui/src/components/grabbable/QasGrabbable.vue

```diff
@@ -0,0 +1,174 @@
+<template>
+  <div class="qas-grabbable relative-position">
+    <div
+      ref="grabContainer"
+      class="flex no-wrap qas-grabbable__container"
+      :class="classes"
+    >
+      <slot />
+    </div>
+  </div>
+</template>
+
+<script setup>
+import { computed, ref, onMounted, onBeforeUnmount } from 'vue'
+import { setScrollOnGrab } from '../../helpers'
+
+defineOptions({ name: 'QasGrabbable' })
+
+const props = defineProps({
+  useScrollBar: {
+    type: Boolean
+  }
+})
+
+const emit = defineEmits(['grabbing'])
+
+const grabContainer = ref(null)
+const grabPosition = ref(null)
+const isGrabbing = ref(false)
+const scrollOnGrab = ref({})
+
+const classes = computed(() => {
+  const baseClass = 'qas-grabbable__container'
+
+  return [
+    {
+      [`${baseClass}--grabbing`]: isGrabbing.value,
+      [`${baseClass}--no-grab`]: !hasScrollOnGrab.value,
+      [`${baseClass}--no-scroll`]: !props.useScrollBar
+    },
+    `${baseClass}--grab-${grabPosition.value}`
+  ]
+})
+
+const hasScrollOnGrab = computed(() => !!Object.keys(scrollOnGrab.value).length)
+
+function handleScrollOnGrab () {
+  const { scrollWidth, offsetWidth } = grabContainer.value
+
+  if (scrollWidth > offsetWidth) {
+    initScrollOnGrab()
+    setGrabPosition()
+
+    return
+  }
+
+  destroyEventsAndResetGrab()
+}
+
+function initScrollOnGrab () {
+  if (hasScrollOnGrab.value) return
+
+  scrollOnGrab.value = setScrollOnGrab(grabContainer.value, {
+    onGrabFn: onGrab,
+    onMoveFn: setGrabPosition,
+    onScrollFn: setGrabPosition
+  })
+}
+
+function destroyEventsAndResetGrab () {
```

Comment:
aqui é onde ele destroy tudo né, será que não faz mais sentido um nome mais generico neste caso? pq senão tivesse mais uma ação, teria mais um "And"?

Reply:
Coloquei o nome de `disableScrollOnGrab`, o que acha?